### ISSUE TITLE
Selective `supabase migration squash`

### ISSUE THREAD

**Issue by henry2man** [supabase/cli#2239]
**Is your feature request related to a problem? Please describe.**
I'm a little bit frustrated when preparing for a production deployment using `supabase migration squash` command because it consolidates all migrations, including those already applied to the production database. This can lead to unnecessary complications and the need to manually select which migrations to include for production environments. We have a `--version` argument but this not let us to select a range of versions, for example.

**Describe the solution you'd like**
I would like the supabase migration squash command to include an option to squash only migrations that have not yet been applied to the remote production database. This would streamline the process of preparing migrations for production by automatically generating a single migration script from the unapplied migrations, thus reducing the risk of errors and manual filtering.

**Describe alternatives you've considered**
An alternative could be manually filtering out the applied migrations and only squashing the remaining ones, but this is error-prone and not scalable for teams or frequent deployments.

**Additional context**
Adding parameters such as `--unapplied-only` to the squash command could be a straightforward solution to ensure that only new, unapplied migrations are included in the squashed file. This would make the migration process much more efficient and error-free for teams moving changes to production environments.

**Comment by henry2man** [comment]
Yes, this is the use case, merging the N last, non applied scripts. 

What if I have applied local scripts? I don't want to mess the migration table.

**Comment by joelpramos** [comment]
@henry2man curious how you solved this in your workflow

**Comment by henry2man** [comment]
> @henry2man curious how you solved this in your workflow


@joelpramos I didn't. I handle all these by hand, reverting and re-applying migrations by hand 🤷‍♂️
